android-animation-arsenal/src/main/java/com/android_animation_arsenal/views/RevealDialog.java

```diff
@@ -0,0 +1,373 @@
+        hasTarget = true;
+    }
+
+    @Override
+    public void show()
+    {
+        if(! isReveal)
+        {
+            if(hasTarget)
+            {
+                targetView.setEnabled(true);
+                targetView.setClickable(false);
+            }
+            getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color
+                    .TRANSPARENT));
+            setOnShowListener(new OnShowListener()
+            {
+                @Override
+                public void onShow(DialogInterface sdialog)
+                {
+                    if(hasTarget)
+                    {
+                        hide();
+                        revealDialogWithTarget();
+                    }
+                    else
+                    {
+                        revealDialog();
+                    }
+                }
+            });
+        }
+        super.show();
+    }
+
+    /**
+     * Reveals dialog without target
+     */
+    public void revealDialog()
+    {
+        int x = contentView.getWidth() / 2;
+        int y = contentView.getHeight() / 2;
+        DisplayMetrics displayMetrics = getContext().getResources().getDisplayMetrics();
+        // big radius to cover view
+        int bigRadius = Math.max(displayMetrics.widthPixels, displayMetrics
+                .heightPixels);
+        Animator toolbarExpandAnim = ViewAnimationUtils.createCircularReveal(
+                contentView, x, y, 0, bigRadius);
+        toolbarExpandAnim.setStartDelay(duration);
+        toolbarExpandAnim.setDuration(duration);
+    }
+
+    @Override
+    public void setContentView(View view)
+    {
+        super.setContentView(view);
+        contentView = view;
+        contentView.setOnTouchListener(new View.OnTouchListener()
+        {
+            @Override
+            public boolean onTouch(View v, MotionEvent event)
+            {
+                return true;
+            }
+        });
+    }
+
+    /**
+     * Animates dialog into a reveal and handles
+     * target animation from its original position to dialog position
+     */
+    public void revealDialogWithTarget()
+    {
+        if(isAnimating)
+        {
+            return;
+        }
+
+        isAnimating = true;
+
+        // get display metrics
+        DisplayMetrics metrics = new DisplayMetrics();
+        getWindow().getWindowManager().getDefaultDisplay().getMetrics(metrics);
+
+        // get center coordinates of dialog
+        final int x = contentView.getWidth() / 2;
+        final int y = contentView.getHeight() / 2;
+
+        // disable targetView
+        targetView.setEnabled(false);
+        targetView.setClickable(false);
+
+        // change alpha of contentView and visibility
+        contentView.setAlpha(0f);
+        contentView.setVisibility(View.VISIBLE);
+
+        // get original position of targetView
+        final int originalPos[] = new int[2];
+        targetView.getLocationOnScreen(originalPos);
```

Comment:
getLocationOnScreen has been used multiple times in the code, probably it will be easier to write a util function that returns an int array with the locations.
`public static int[]  getLocationOnScreen(View view)`